docker/ubuntu_18.04-dev.dockerfile

```diff
@@ -58,50 +58,67 @@ RUN if [ "${py_version%.?}" -eq 3 ] ; \
             future
 
 
+
+
 # make starting directory
-ARG build_moab=NO
-ARG enable_pymoab=NO
 RUN mkdir -p $HOME/opt
 RUN echo "export PATH=$HOME/.local/bin:\$PATH" >> ~/.bashrc
 
+# build HDF5
+ARG build_hdf5="NO"
+ENV HDF5_INSTALL_PATH=$HOME/opt/hdf5/$build_hdf5
+RUN if [ "$build_hdf5" != "NO" ]; then \
+        cd $HOME/opt \
+        && mkdir hdf5 \
+        && cd hdf5 \
+        && git clone --single-branch --branch $build_hdf5 https://github.com/HDFGroup/hdf5.git \
+        && cd hdf5 \
+        && ./configure --prefix=$HDF5_INSTALL_PATH --enable-shared \
+        && make -j 3 \
+        && make install \
+        && cd .. \
+        && rm -rf hdf5; \
+    fi
+# put MOAB on the path
+ENV LD_LIBRARY_PATH $HDF5_INSTALL_PATH/lib:$LD_LIBRARY_PATH
+ENV LIBRARY_PATH $HDF5_INSTALL_PATH/lib:$LIBRARY_PATH
+
+ARG build_moab=NO
+ARG enable_pymoab=NO
+ENV INSTALL_PATH=$HOME/opt/moab
```

Comment:
I believe than ARG command must be placed just before line they are used, 
I am unsure if having a ENV in-between could be a problem.

you might need to redeclare `build_hdf5` as an argument there along with `build_moab` and `enable_moab`

Reply:
This is not a problem - everything works as it is....

Reply:
I had to make sure...

I was convince that during one of my previous experience with docker `ARGS` I add to read the `ARG MY_VAR=XX` before each command that was using the variable....

I tested and apparently not...